src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -1170,6 +1170,20 @@ bool HMICapabilitiesImpl::load_capabilities_from_file() {
           smart_objects::SmartObject rc_capability_so;
           formatters::CFormatterJsonBase::jsonValueToObj(rc_capability,
                                                          rc_capability_so);
+          if (rc_capability_so.keyExists("lightControlCapabilities")) {
+            if (rc_capability_so["lightControlCapabilities"].keyExists(
+                    "supportedLights")) {
+              auto& lights = rc_capability_so["lightControlCapabilities"]
+                                             ["supportedLights"];
+              auto it = lights.asArray()->begin();
+              for (; it != lights.asArray()->end(); ++it) {
+                smart_objects::SmartObject& light_name_so = (*it)["name"];
+                auto light_name = MessageHelper::CommonLightNameFromString(
```

Comment:
@LuxoftAKutsan please see Jacobs comment above

Reply:
@LuxoftAKutsan That is not true, Buttons are controlled in the same way, there is just a workaround in the `ButtonPress` code to handle if this value is a string https://github.com/smartdevicelink/sdl_core/blob/master/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc#L123. A proper solution to this would be to validate these capabilities using the API schema. I will accept this fix for now, but I will also submit an issue relating to how these capabilities should be validated.

Reply:
Created issue: https://github.com/smartdevicelink/sdl_core/issues/2831